src/DemoSat2/work/interruptproxy/implem/default/C/src/interruptproxy.c

```diff
@@ -0,0 +1,25 @@
+/* Body file for function InterruptProxy
+ * Generated by TASTE (kazoo/templates/skeletons/c-body/function.tmplt)
+ * You can edit this file, it will not be overwritten
+
+    !! IMPORTANT                                                        !!
+    !! When you modify your design (interface view), you must update    !!
+    !! the procedures corresponding to the provided interfaces in this  !!
+    !! file. The up-to-date signatures can be found in the .ads file.   !!
+*/
+#include "interruptproxy.h"
+
+void interruptproxy_startup(void)
+{
+
+}
+
+void interruptproxy_PI_InterruptSubscription_Interrupt_Ri
+      (const asn1SccInterruptNumber *IN_interrupt)
+
+{
+   interruptproxy_RI_InterruptSubscription_Interrupt_Pi_Uart(IN_interrupt);
```

Comment:
@Lurkerpas maybe it would be a good idea to call the given interface only when the proper interrupt was called. Thus, enqueuing fewe false messages for AFEC and UART.
```suggestion
   if(*IN_interrupt >= Nvic_Irq_Uart0 && *IN_interrupt <= Nvic_Irq_Uart4)
       interruptproxy_RI_InterruptSubscription_Interrupt_Pi_Uart(IN_interrupt);
```

Reply:
Done